Allow empty initial time when using text input mode in showTimePicker dialog

azatech opened the pull request:
Added ability to allow empty initial time when using text input mode in showTimePicker dialog #169131 

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

azatech committed:
Allow empty initial time when using text input mode in showTimePicker dialog

MitchellGoodwin reviewed (commented):
Thank you for putting this together. It's a little awkward when using this code as the initial date still shows as a placeholder, which might be confusing. I think the hint text for the field can be set to show nothing if this new property is true:
<img width="414" alt="Screenshot 2025-07-01 at 3 08 26 PM" src="https://github.com/user-attachments/assets/53f0b7b9-29b9-4245-ba25-c9538b542dc6" />

azatech committed:
Merge branch 'master' into 169131_allow-empty-initial-time-when-using-text-input-mode-in-show-time-picker

azatech committed:
Allow empty initial time when using text input mode in showTimePicker dialog

MitchellGoodwin commented on packages/flutter/lib/src/material/time_picker.dart:
```diff
@@ -2274,6 +2303,13 @@ class TimePickerDialog extends StatefulWidget {
+  /// If true and entry mode is [TimePickerEntryMode.input], the hour and minute
+  /// fields will be empty on start instead of pre-filled with [initialTime].
+  ///
+  /// Improves UX by removing the need to delete default values before typing.
+  /// Skipped in dial mode.
```
```suggestion
  /// Has no effect in dial mode.
```
Nit: Generally we try and avoid putting opinions into the documentation like, "do this for good UX".

MitchellGoodwin commented on packages/flutter/lib/src/material/time_picker.dart:
```diff
@@ -2107,7 +2128,14 @@ class _HourMinuteTextFieldState extends State<_HourMinuteTextField> with Restora
     // Remove the hint text when focused because the centered cursor
     // appears odd above the hint text.
-    final String? hintText = focusNode.hasFocus ? null : _formattedValue;
+    // Clear the hint text when emptyInitialTime is true to avoid showing a
+    // misleading placeholder.
```
```suggestion
    // Clear the hint text when emptyInitialTime is true.
```

MitchellGoodwin commented on packages/flutter/lib/src/material/time_picker.dart:
```diff
@@ -2107,7 +2128,14 @@ class _HourMinuteTextFieldState extends State<_HourMinuteTextField> with Restora
-    final String? hintText = focusNode.hasFocus ? null : _formattedValue;
+    // Clear the hint text when emptyInitialTime is true to avoid showing a
+    // misleading placeholder.
+    final String? hintText =
+        focusNode.hasFocus
+            ? null
+            : widget.emptyInitialTime
+            ? ''
+            : _formattedValue;
```
```suggestion
    final String? hintText =
        focusNode.hasFocus || widget.emptyInitialTime
            ? null
            : _formattedValue;
```
I think this would do the same thing and be a little cleaner?

azatech replied:
Yes, that would be cleaner. I've just adjusted code! 
Thank you @MitchellGoodwin

MitchellGoodwin reviewed (commented):
Thank you for the update. I left a few small nits.

azatech committed:
Merge remote-tracking branch 'origin/master' into 169131_allow-empty-initial-time-when-using-text-input-mode-in-show-time-picker

azatech committed:
Code, comments and tests enhancements

azatech committed:
Update packages/flutter/lib/src/material/time_picker.dart

Enhanced comments

Co-authored-by: Mitchell Goodwin <58190796+MitchellGoodwin@users.noreply.github.com>

azatech committed:
Merge remote-tracking branch 'origin/169131_allow-empty-initial-time-when-using-text-input-mode-in-show-time-picker' into 169131_allow-empty-initial-time-when-using-text-input-mode-in-show-time-picker

# Conflicts:
#	packages/flutter/lib/src/material/time_picker.dart

azatech commented:
Thanks @MitchellGoodwin for the review. All minor fixes done! 
We are ready to go ;)

mdebbar committed:
Revert "Mark web_long_running_tests_2_5 as bringup" (#171872)

Reverts flutter/flutter#171726

The failure is fixed in https://github.com/flutter/flutter/pull/171801
and the task has been green ever since.

<img width="275" alt="image"
src="https://github.com/user-attachments/assets/f2fc18e6-7dcd-4017-a5c5-ade6988b0692"
/>

Follow up for https://github.com/flutter/flutter/issues/171725

ahmedsameha1 committed:
Replace [FinderBase] with [Finder] in the documentation of Matchers (#168279)

Replace [FinderBase] with [Finder] in the doc comments of Matchers.

Closes #168230

engine-flutter-autoroll committed:
Roll Packages from cba2e9028121 to 4a231ae8546d (5 revisions) (#171879)

https://github.com/flutter/packages/compare/cba2e9028121...4a231ae8546d

2025-07-08 stuartmorgan@google.com [video_player] Adds platform view
support on macOS (flutter/packages#9576)
2025-07-08 tirth@nevercode.io [camera] fix `CameraLensType` export
(flutter/packages#9536)
2025-07-08 stuartmorgan@google.com [video_player] Move iOS/macOS to
per-player-instance Pigeon APIs (flutter/packages#9529)
2025-07-08 engine-flutter-autoroll@skia.org Roll Flutter from
28a4e852bc90 to adffe244f3f1 (17 revisions) (flutter/packages#9580)
2025-07-08 stuartmorgan@google.com [google_sign_in] Don't crash a
misconfigured iOS app (flutter/packages#9486)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-packages-flutter-autoroll
Please CC flutter-ecosystem@google.com on the revert to ensure that a
human
is aware of the problem.

To file a bug in Flutter:
https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

kjlubick committed:
[skia] Fix flag fiddling for Fuchsia, FreeType, & friends (#171874)

In https://review.skia.org/1017296, Skia deleted some logic in skia.gni
that dealt with Fuchsia. Flutter had been unintentionally depending on
that to set FreeType integration, as well as Ganesh GL/Vulkan and ICU.
This moves those flags into Flutter's BUILD.gn file for Skia, making the
explicit and keeping the old behavior the same as the new behavior.

I had to add an explicit dependency between the Fuchsia SkFontMgr and
Freetype, which has been "just working" by accident before.

See https://g-issues.skia.org/issues/427681157 for more.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description
above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel
on [Discord].

<!-- Links -->
[Contributor Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

auto-submit[bot] committed:
Reverts "Roll Packages from cba2e9028121 to 4a231ae8546d (5 revisions) (#171879)" (#171897)

<!-- start_original_pr_link -->
Reverts: flutter/flutter#171879
<!-- end_original_pr_link -->
<!-- start_initiating_author -->
Initiated by: chinmaygarde
<!-- end_initiating_author -->
<!-- start_revert_reason -->
Reason for reverting: https://github.com/flutter/flutter/issues/171896
<!-- end_revert_reason -->
<!-- start_original_pr_author -->
Original PR Author: engine-flutter-autoroll
<!-- end_original_pr_author -->

<!-- start_reviewers -->
Reviewed By: {fluttergithubbot}
<!-- end_reviewers -->

<!-- start_revert_body -->
This change reverts the following previous change:

https://github.com/flutter/packages/compare/cba2e9028121...4a231ae8546d

2025-07-08 stuartmorgan@google.com [video_player] Adds platform view
support on macOS (flutter/packages#9576)
2025-07-08 tirth@nevercode.io [camera] fix `CameraLensType` export
(flutter/packages#9536)
2025-07-08 stuartmorgan@google.com [video_player] Move iOS/macOS to
per-player-instance Pigeon APIs (flutter/packages#9529)
2025-07-08 engine-flutter-autoroll@skia.org Roll Flutter from
28a4e852bc90 to adffe244f3f1 (17 revisions) (flutter/packages#9580)
2025-07-08 stuartmorgan@google.com [google_sign_in] Don't crash a
misconfigured iOS app (flutter/packages#9486)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-packages-flutter-autoroll
Please CC flutter-ecosystem@google.com on the revert to ensure that a
human
is aware of the problem.

To file a bug in Flutter:
https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

<!-- end_revert_body -->

Co-authored-by: auto-submit[bot] <flutter-engprod-team@google.com>

auto-submit[bot] committed:
Reverts "Reverts "Roll Packages from cba2e9028121 to 4a231ae8546d (5 revisions) (#171879)" (#171897)" (#171910)

<!-- start_original_pr_link -->
Reverts: flutter/flutter#171897
<!-- end_original_pr_link -->
<!-- start_initiating_author -->
Initiated by: matanlurey
<!-- end_initiating_author -->
<!-- start_revert_reason -->
Reason for reverting: Was not the reason for the test failure, it was an
OOB tag change.
<!-- end_revert_reason -->
<!-- start_original_pr_author -->
Original PR Author: auto-submit[bot]
<!-- end_original_pr_author -->

<!-- start_reviewers -->
Reviewed By: {fluttergithubbot}
<!-- end_reviewers -->

<!-- start_revert_body -->
This change reverts the following previous change:
<!-- start_original_pr_link -->
Reverts: flutter/flutter#171879
<!-- end_original_pr_link -->
<!-- start_initiating_author -->
Initiated by: chinmaygarde
<!-- end_initiating_author -->
<!-- start_revert_reason -->
Reason for reverting: https://github.com/flutter/flutter/issues/171896
<!-- end_revert_reason -->
<!-- start_original_pr_author -->
Original PR Author: engine-flutter-autoroll
<!-- end_original_pr_author -->

<!-- start_reviewers -->
Reviewed By: {fluttergithubbot}
<!-- end_reviewers -->

<!-- start_revert_body -->
This change reverts the following previous change:

https://github.com/flutter/packages/compare/cba2e9028121...4a231ae8546d

2025-07-08 stuartmorgan@google.com [video_player] Adds platform view
support on macOS (flutter/packages#9576)
2025-07-08 tirth@nevercode.io [camera] fix `CameraLensType` export
(flutter/packages#9536)
2025-07-08 stuartmorgan@google.com [video_player] Move iOS/macOS to
per-player-instance Pigeon APIs (flutter/packages#9529)
2025-07-08 engine-flutter-autoroll@skia.org Roll Flutter from
28a4e852bc90 to adffe244f3f1 (17 revisions) (flutter/packages#9580)
2025-07-08 stuartmorgan@google.com [google_sign_in] Don't crash a
misconfigured iOS app (flutter/packages#9486)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/flutter-packages-flutter-autoroll
Please CC flutter-ecosystem@google.com on the revert to ensure that a
human
is aware of the problem.

To file a bug in Flutter:
https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

<!-- end_revert_body -->

<!-- end_revert_body -->

Co-authored-by: auto-submit[bot] <flutter-engprod-team@google.com>

matanlurey committed:
Add public postmortem of the 3.32.3 release. (#171904)

matanlurey committed:
Make `labels` field an array (#171906)

Maybe it works both ways, but just to make sure.

nt4f04uNd committed:
Add detailed error message for BorderRadiusDirectional (#171805)

<!--
Thanks for filing a pull request!
Reviewers are typically assigned within a week of filing a request.
To learn more about code review, see our documentation on Tree Hygiene:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
-->

This PR adds a more descriptive error message for cases when
BorderRadiusDirectional is used in widgets whose ancestors aren't
wrapped into Directionality.

Fixes https://github.com/flutter/flutter/issues/88644

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description
above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel
on [Discord].

<!-- Links -->
[Contributor Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

matanlurey committed:
Simplify the template for infrastructure requests (#171905)

Closes https://github.com/flutter/flutter/issues/165814.

matanlurey committed:
Add total execution time to the flutter upgrade command (#171475)

Closes https://github.com/flutter/flutter/issues/47090.

jtmcdole committed:
feat: new builders for size experiment (#171886)

These builders are experimental and should not be used in production.

copied from ddm experiment

matanlurey committed:
Update `.gitignore`s (#171907)

Closes https://github.com/flutter/flutter/issues/54687.

biggs0125 committed:
Add analytics events for wasm dry runs on web builds (#171818)

Follow up to https://github.com/flutter/flutter/pull/171682
Depends on https://github.com/dart-lang/tools/pull/2125

---------

Co-authored-by: Nate Biggs <natebiggs@google.com>

jesswrd committed:
[Android 16] Added Docs to Warn Users that SystemChrome.setPreferredOrientations will Not Work (#171089)

Added docs to warn users that `SystemChrome.setPreferredOrientations`
will not work when Flutter apps target Android 16 (SDK 36).

Fixes https://github.com/flutter/flutter/issues/162724

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description
above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel
on [Discord].

<!-- Links -->
[Contributor Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

matanlurey committed:
Support `NO_COLOR` to opt-out of `flutter` tool ANSI colors (#171892)

Closes https://github.com/flutter/flutter/issues/49444.

matanlurey committed:
Hide the rarely direct used `--sample` argument by default (#171898)

Closes https://github.com/flutter/flutter/issues/51859.

dannyvalentesonos committed:
Allow flutter attach to discover flutter engine running on Custom Device (#170635)

## Summary

A Custom Device's log reader doesn't do anything, and thus `flutter
attach` will fail to ever find the flutter engine's VM service.

This change adds a new `readLogs` command to the CustomDeviceConfig, and
which allows a custom device to read the logs from the running flutter
app on the actual custom device, and then allow flutter attach to work.

Fixes #170634

## Testing

Created a custom device, and added the following readLogs command in the
`custom_devices.json` file:
```
"readLogs": [
        "sshpass",
        "-f",
        "/home/dvalente/player.txt",
        "ssh",
        "root@10.0.0.20",
        "tail -Fn +1 /opt/log/flutter.log"
      ]
```

Then, running `flutter attach` works every time.

**Tested the following scenarios:**
* Launched `flutter attach` before app is running on device, and once
app runs, flutter attach succeeds
* Launched `flutter attach` after app is already running on device, and
flutter attach succeeds.
* Used `flutter attach` in Android Studio, and hot reload works just
like any Android device.

matanlurey committed:
Add `flutter config --enable-omit-legacy-version-file` (#171903)

Towards https://github.com/flutter/flutter/issues/171900.

We can decide how quickly to move here, but a feature flag will make it
easier to try out/rollback.

camsim99 committed:
Update `CHANGELOG` for 3.32.5, 3.32.6 stable hotfix releases (#171891)

Updates `CHANGELOG` for 3.32.5, 3.32.6 stable hotfix releases.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description
above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel
on [Discord].

<!-- Links -->
[Contributor Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

---------

Co-authored-by: Reid Baker <1063596+reidbaker@users.noreply.github.com>

mboetger committed:
Use Async SurfaceHolder Callback to remove need for setting alpha workaround (#171398)

We can rely on the SurfaceHolder.Callback2 method,
surfaceRedrawNeededAsync to properly signal the below Surface that
rendering of the frame is done, and the Surface can be finished drawing.
This removes the need to set the alpha to prevent the blank surface from
being rendered before the engine is done.

Fixes: #140246

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description
above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [X] All existing and new tests are passing.

gaaclarke committed:
Adds a MCP server for working with the engine (#171738)

WIP this just adds a few commands and sets up the work for others to
build off of.

Blatant omissions:
1) ~~integration with the engine dart monorepo workspace~~
1) ~~brings in `mcp_dart` dependency (it has an MIT license but is very
new)~~
1) CI testing integration (unless there is some magic that crawls
through the repo and calls them)
1) provide progress notifications and compilation errors to the client
when building
1) `et run`
1) `et test`

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description
above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel
on [Discord].

<!-- Links -->
[Contributor Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

yiiim committed:
Fix the issue with `SliverMainAxisGroups` growing in the reverse direction during layout. (#171005)

Fixes: #170991

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description
above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel
on [Discord].

<!-- Links -->
[Contributor Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

matanlurey committed:
Give an actionable error to `flutter_test.*tap` of a `RenderSliver` (#171930)

Closes https://github.com/flutter/flutter/issues/67743.

robert-ancell committed:
Refactor compositor classes (#171414)

Present frames directly to each view, which maintains its own
compositor.
This simplifies the compositor so it doesn't need to maintain a map from
views to buffers.

engine-flutter-autoroll committed:
Roll Fuchsia Linux SDK from 0-xqmXWc4cXzw3tfe... to lO64ePNEGrGzs-MFC... (#171937)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/fuchsia-linux-sdk-flutter
Please CC chinmaygarde@google.com,zra@google.com on the revert to ensure
that a human
is aware of the problem.

To file a bug in Flutter:
https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

yiiim committed:
Fix the hitTest issue of reversed SliverMainAxisGroup. (#171073)

Fixes: #170999
Fixes: #171121

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description
above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel
on [Discord].

<!-- Links -->
[Contributor Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

ksokolovskyi committed:
[web] Add frame number support. (#171592)

Fixes https://github.com/flutter/flutter/issues/170972

### Description
- Adds `frameData` with `frameNumber` value to `FrameService`
- Adds non-mock `frameData` to `EngineFlutterWindow` and
`EnginePlatformDispatcher`
- Adds `frameNumber` value to `FrameTimingRecorder`'s recorded timings

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description
above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [X] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel
on [Discord].

<!-- Links -->
[Contributor Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]:
https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

mdebbar committed:
Run stateless_stateful_hot_reload_web_test.dart on Mac/Windows (#171283)

Part of https://github.com/flutter/flutter/issues/171276

engine-flutter-autoroll committed:
Roll Skia from 0fef076beec3 to bdb8bfcde7f3 (34 revisions) (#171964)

Roll Skia from 0fef076beec3 to bdb8bfcde7f3 (34 revisions)

https://skia.googlesource.com/skia.git/+log/0fef076beec3..bdb8bfcde7f3

2025-07-10 michaelludwig@google.com [graphite] Improve no-op scissor
handling
2025-07-10 kjlubick@google.com Revert "Delete GN support for building
for fuchsia"
2025-07-10 lehoangquyen@chromium.org GraphiteDawn: Use
@interpolate(flat, either) for BlitWithDraw pipeline
2025-07-10 nicolettep@google.com [graphite] DrawShadow optimizations
2025-07-10 kjlubick@google.com Temporarily remove Radeon Macbook Pro
Tests from CQ
2025-07-10 kjlubick@google.com Enforce IWYU on more graphite files
2025-07-10 skia-autoroll@skia-public.iam.gserviceaccount.com Roll ANGLE
from ce289330b0b8 to ec2a04cc9535 (11 revisions)
2025-07-10 skia-autoroll@skia-public.iam.gserviceaccount.com Roll
SwiftShader from be6ed66fa563 to c1f7fbbec4e1 (1 revision)
2025-07-10 skia-autoroll@skia-public.iam.gserviceaccount.com Roll Skia
Infra from 86bd00388c3e to c4debb111e10 (7 revisions)
2025-07-10 skia-autoroll@skia-public.iam.gserviceaccount.com Roll Dawn
from de887f5d1721 to ba8b1ec2a10a (20 revisions)
2025-07-10
recipe-mega-autoroller@chops-service-accounts.iam.gserviceaccount.com
Roll recipe dependencies (trivial).
2025-07-10 skia-autoroll@skia-public.iam.gserviceaccount.com Roll
vulkan-deps from c91d04c5a3ae to 9af2a11b445f (8 revisions)
2025-07-10 thomsmit@google.com Add upper limit to Triangulator contour
counts
2025-07-09 michaelludwig@google.com [graphite] Add Shape::isFloodFill()
helper
2025-07-09 robertphillips@google.com [graphite] Upstream Android
Mourimap changes
2025-07-09 fmalita@google.com Remove SkClipStack use in GrAppliedClip.h
2025-07-09 kjlubick@google.com Remove old flags in skia.gni
2025-07-09 kjlubick@google.com Remove uses of deprecated no-args
makeRasterImage
2025-07-09 skia-autoroll@skia-public.iam.gserviceaccount.com Roll
vulkan-deps from e573f4bc7407 to c91d04c5a3ae (4 revisions)
2025-07-09
recipe-mega-autoroller@chops-service-accounts.iam.gserviceaccount.com
Roll recipe dependencies (trivial).
2025-07-09 skia-autoroll@skia-public.iam.gserviceaccount.com Roll ANGLE
from 825b40622cf2 to ce289330b0b8 (5 revisions)
2025-07-09 skia-autoroll@skia-public.iam.gserviceaccount.com Roll Skia
Infra from 743562b36d09 to 86bd00388c3e (15 revisions)
2025-07-09 skia-autoroll@skia-public.iam.gserviceaccount.com Roll Dawn
from d7b31b22748d to de887f5d1721 (17 revisions)
2025-07-08 skia-autoroll@skia-public.iam.gserviceaccount.com Roll
vulkan-deps from 242b87cc5bcc to e573f4bc7407 (6 revisions)
2025-07-08 borenet@google.com [infra] skia-cdn has moved to
skia-world-readable
2025-07-08 skia-autoroll@skia-public.iam.gserviceaccount.com Manual roll
Dawn from 0275c582a080 to d7b31b22748d (11 revisions)
2025-07-08 michaelludwig@google.com [graphite] Define inner bounds for
draws when combing with clip stack
2025-07-08 xing.xu@intel.com GraphiteDawn: Use dawn partial resolve
feature on D3D11
2025-07-08 robertphillips@google.com [graphite] Add a PaintOptions
Builder
2025-07-08
recipe-mega-autoroller@chops-service-accounts.iam.gserviceaccount.com
Roll recipe dependencies (trivial).
2025-07-08 michaelludwig@google.com [graphite] Add debugging members for
ResourceCache usage
2025-07-08 kjlubick@google.com Delete GN support for building for
fuchsia
2025-07-08 kjlubick@google.com Generate graphite.gni from BUILD.bazel
2025-07-08 thomsmit@google.com [graphite] Add more DrawAtlas test
coverage

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/skia-flutter-autoroll
Please CC
chinmaygarde@google.com,kjlubick@google.com,thomsmit@google.com on the
revert to ensure that a human
is aware of the problem.

To file a bug in Skia: https://bugs.chromium.org/p/skia/issues/entry
To file a bug in Flutter:
https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622
...

rkishan516 committed:
Feat: Add foreground color for cupertino button (#170898)

Feat: Add foreground color for cupertino button
fixes: #169507 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description
above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [x] All existing and new tests are passing.

---------

Co-authored-by: Tong Mu <dkwingsmt@users.noreply.github.com>

engine-flutter-autoroll committed:
Roll Skia from bdb8bfcde7f3 to dc3da09ca905 (3 revisions) (#171971)

https://skia.googlesource.com/skia.git/+log/bdb8bfcde7f3..dc3da09ca905

2025-07-10 kjlubick@google.com Fix compile errors in Graphite's Vello
code
2025-07-10 robertphillips@google.com [graphite] Remove trampoline
functions from Precompilation tests
2025-07-10 skia-autoroll@skia-public.iam.gserviceaccount.com Roll
vulkan-deps from 9af2a11b445f to 3e4d34bfa361 (3 revisions)

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/skia-flutter-autoroll
Please CC
chinmaygarde@google.com,kjlubick@google.com,thomsmit@google.com on the
revert to ensure that a human
is aware of the problem.

To file a bug in Skia: https://bugs.chromium.org/p/skia/issues/entry
To file a bug in Flutter:
https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

rkishan516 committed:
Style: Rename pageBuilder with builder in showCupertinoSheet (#170625)

Style: Rename pageBuilder with builder in showCupertinoSheet
fixes: #169831

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined
there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my
responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features
we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description
above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is
[test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven
Fixes] where supported.
- [x] All existing and new tests are passing.

MitchellGoodwin reviewed (approved):
LGTM, but the ci.yaml is getting stuck on something, and I'm unable to rebase through github. Could you rebase this PR? Then I'll ask around for a second review.

jason-simmons committed:
Manual roll Dart SDK from 8d69b07b9d9d to 07ea3aaaadf0 (32 revisions) (#171969)

https://dart.googlesource.com/sdk.git/+log/8d69b07b9d9d..07ea3aaaadf0

2025-07-09 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-327.0.dev
2025-07-09 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-326.0.dev
2025-07-09 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-325.0.dev
2025-07-09 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-324.0.dev
2025-07-09 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-323.0.dev
2025-07-09 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-322.0.dev
2025-07-09 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-321.0.dev
2025-07-08 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-320.0.dev
2025-07-08 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-319.0.dev
2025-07-08 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-318.0.dev
2025-07-08 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-317.0.dev
2025-07-07 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-316.0.dev
2025-07-07 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-315.0.dev
2025-07-05 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-314.0.dev
2025-07-04 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-313.0.dev
2025-07-04 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-312.0.dev
2025-07-04 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-311.0.dev
2025-07-04 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-310.0.dev
2025-07-03 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-309.0.dev
2025-07-03 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-308.0.dev
2025-07-03 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-307.0.dev
2025-07-03 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-306.0.dev
2025-07-03 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-305.0.dev
2025-07-03 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-304.0.dev
2025-07-02 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-303.0.dev
2025-07-02 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-302.0.dev
2025-07-02 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-301.0.dev
2025-07-02 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-300.0.dev
2025-07-02 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-299.0.dev
2025-07-02 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-298.0.dev
2025-07-01 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-297.0.dev
2025-07-01 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-296.0.dev
2025-07-01 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-295.0.dev
2025-07-01 dart-internal-merge@dart-ci-internal.iam.gserviceaccount.com
Version 3.9.0-294.0.dev

If this roll has caused a breakage, revert this CL and stop the roller
using the controls here:
https://autoroll.skia.org/r/dart-sdk-flutter
Please CC chinmaygarde@google.com,dart-vm-team@google.com on the revert
to ensure that a human is aware of the problem.

To file a bug in Flutter:
https://github.com/flutter/flutter/issues/new/choose

To report a problem with the AutoRoller itself, please file a bug:
https://issues.skia.org/issues/new?component=1389291&template=1850622

Documentation for the AutoRoller is here:
https://skia.googlesource.com/buildbot/+doc/main/autoroll/README.md

---------

Co-authored-by: engine-flutter-autoroll <engine-flutter-autoroll@skia.org>

azatech committed:
Merge branch 'master' into 169131_allow-empty-initial-time-when-using-text-input-mode-in-show-time-picker

dkwingsmt committed:
Merge branch 'master' into 169131_allow-empty-initial-time-when-using-text-input-mode-in-show-time-picker

MitchellGoodwin commented:
@azatech the commit history on this PR has gotten pretty crazy, and has pulled in a bunch of unrelated commits. Can you try and remove all commits that aren't yours? It might be easier to open a new PR with these commits. If so, tag me and I'll approve that one as well and we'll go from there.

dkwingsmt committed:
Merge branch 'master' into 169131_allow-empty-initial-time-when-using-text-input-mode-in-show-time-picker
